control/control.go

```diff
@@ -366,14 +368,16 @@ func (c *Controller) Status(req *controlapi.StatusRequest, stream controlapi.Con
 						Msg:       v.Data,
 						Timestamp: v.Timestamp,
 					})
-					logSize += len(v.Data)
+					logSize += len(v.Data) + emtpyLogVertexSize
 					// avoid logs growing big and split apart if they do
 					if logSize > 1024*1024 {
 						ss.Vertexes = nil
 						ss.Statuses = nil
 						ss.Logs = ss.Logs[i+1:]
 						retry = true
 						break
+					}else {
+						retry = false
```

Comment:
why this?

Reply:
All items after first message bigger than 1M will be sent separately if not set retry back to false.

Reply:
after first big message the retry always be true and there is an endless loop.

Reply:
hmm, looks weird indeed. maybe the `retry := false` should be inside the `for`

Reply:
fixed